File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0, right = 0; 
        int res = 0; 
        while(right < prices.length){
            int profit = prices[right] - prices[left];
            if(profit < 0){
                left = right; 
            }
            res = Math.max(res, profit);
            right += 1; 
        }

        return res; 
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left, right = 0, 0
        counter = 0 
        visited = set() 
        while right < len(s): 
            while s[right] in visited:
                visited.remove(s[left])
                left += 1 

            visited.add(s[right])
            counter = max(counter, right - left + 1)
            right += 1

        return counter 